.eslintrc.js

```diff
@@ -47,7 +51,36 @@ module.exports = {
       plugins: ['node'],
       rules: Object.assign({}, require('eslint-plugin-node').configs.recommended.rules, {
         // add your custom rules and overrides for node files here
+        "@typescript-eslint/explicit-function-return-type": ["off"],
+        "@typescript-eslint/no-var-requires": ["off"],
+        "no-dupe-else-if": ["off"],
+        "no-import-assign": ["off"],
+        "no-setter-return": ["off"],
+        "@typescript-eslint/camelcase": ["off"]
       })
-    }
+    },
+    {
+      files: ["*.ts"],
+      rules: {
+        "@typescript-eslint/explicit-function-return-type": [
+          "error",
+          {
+            allowExpressions: true,
+            allowHigherOrderFunctions: true
+          }
+        ],
+        "no-dupe-else-if": ["off"],
+        "no-import-assign": ["off"],
+        "no-setter-return": ["off"],
+      }
+    },
+    {
+      files: ["tests/dummy/app/router.js"],
```

Comment:
Why do we need this?

Reply:
On reflection, we actually *can* make it `router.ts` (though we could also just delete it for this scenario I think)—we don't in general touch it because the Ember generator doesn't work with the router b/c of how it does the transform.

Reply:
Sure but I more meant that I don't think we need this override at all (I think once you hoist the `no-dupe-else-if`, `no-import-assign`, `no-setter-return` up to the top `rules` you won't need this at all).

Reply:
Oh, yes, agreed. Meant to say as much but was typing and playing D&D simultaneously. 😂